src/qibolab/instruments/qblox/controller.py

```diff
@@ -189,25 +189,24 @@ def _execute_pulse_sequence(
                 and not module_pulses[name].ro_pulses.is_empty
             ):
                 results = module.acquire()
-                existing_keys = set(acquisition_results.keys()) & set(results.keys())
                 for key, value in results.items():
-                    if key in existing_keys:
-                        acquisition_results[key].update(value)
-                    else:
-                        acquisition_results[key] = value
-
+                    acquisition_results[key] = value
         # TODO: move to QRM_RF.acquire()
         shape = tuple(len(sweeper.values) for sweeper in reversed(sweepers))
         shots_shape = (nshots,) + shape
         for ro_pulse in sequence.ro_pulses:
             if options.acquisition_type is AcquisitionType.DISCRIMINATION:
-                _res = acquisition_results[ro_pulse.serial][2]
+                _res = acquisition_results[ro_pulse.serial].classified
                 _res = np.reshape(_res, shots_shape)
                 if options.averaging_mode is not AveragingMode.SINGLESHOT:
                     _res = np.mean(_res, axis=0)
+            elif options.acquisition_type is AcquisitionType.RAW:
+                i_raw = acquisition_results[ro_pulse.serial].raw_i
+                q_raw = acquisition_results[ro_pulse.serial].raw_q
+                _res = i_raw + 1j * q_raw
             else:
```

Comment:
Btw, I would explicitly check for `AcquisitionType.INTEGRATION` here, and fail otherwise (I would do in any case, but consider there is also the `SPECTROSCOPY` type, that should be a Zurich specific mode...)

Reply:
Solved in [eff2153](https://github.com/qiboteam/qibolab/pull/728/commits/eff2153bb6b625f89f7d974ca270d61fe6318d26).